opal/mca/btl/usnic/btl_usnic_module.c

```diff
@@ -1777,6 +1780,35 @@ static void init_find_transport_header_len(opal_btl_usnic_module_t *module)
         module->fabric_info->ep_attr->msg_prefix_size;
     mca_btl_usnic_component.transport_protocol =
         module->fabric_info->ep_attr->protocol;
+
+    /* The usnic provider in libfabric v1.0.0 (i.e., API v1.0) treated
+       FI_MSG_PREFIX inconsistently between senders and receivers.  It
+       was corrected in libfabric v1.1.0 (i.e., API v1.1), meaning
+       that FI_MSG_PREFIX is treated consistently between senders and
+       receivers.
+
+       So check what version of the libfabric API we have, and setup
+       to use the "old" (inconsistent) MSG_PREFIX behavior, or the
+       "new" MSG_PREFIX (consistent) behavior.
+
+       NOTE: This is a little redundant; we're setting a
+       component-level attribute during each module's setup.  We do
+       this here (and not earlier, when we check fi_version() during
+       the component setup) because we can't obtain the value of the
+       endpoint msg_prefix_size until we setup the first module.
+       Also, it's safe because each module will set the component
+       attribute to the same value.  So it's ok. */
+    uint32_t libfabric_api;
+    libfabric_api = fi_version();
+    if (1 == FI_MAJOR(libfabric_api) &&
+        0 == FI_MINOR(libfabric_api)) {
+        // Old MSG_PREFIX behavior: offset = 0
```

Comment:
Consider striking the bit after the `:`, since the code below it is extremely clear and this comment can only end up getting out of date / confusing later.  Or kill the whole comment, given the novella written above this whole stanza.

Ditto for the `else` comment.